In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

data_frame=pd.read_csv(r"C:\Users\AASHIGA\Downloads\archive (3)\amazon.csv")



In [4]:
data_frame.shape

(1465, 16)

In [5]:
display(data_frame.head())

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,about_product,user_id,user_name,review_id,review_title,review_content,img_link,product_link
0,B07JW9H4J1,Wayona Nylon Braided USB to Lightning Fast Cha...,Computers&Accessories|Accessories&Peripherals|...,₹399,"₹1,099",64%,4.2,"24,269",High Compatibility : Compatible With iPhone 12...,"AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...","Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...","R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...","Satisfied,Charging is really fast,Value for mo...",Looks durable Charging is fine tooNo complains...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...
1,B098NS6PVG,Ambrane Unbreakable 60W / 3A Fast Charging 1.5...,Computers&Accessories|Accessories&Peripherals|...,₹199,₹349,43%,4.0,"43,994","Compatible with all Type C enabled devices, be...","AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...","ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...","RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...","A Good Braided Cable for Your Type C Device,Go...",I ordered this cable to connect my phone to An...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Ambrane-Unbreakable-Char...
2,B096MSW6CT,Sounce Fast Phone Charging Cable & Data Sync U...,Computers&Accessories|Accessories&Peripherals|...,₹199,"₹1,899",90%,3.9,"7,928",【 Fast Charger& Data Sync】-With built-in safet...,"AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...","Kunal,Himanshu,viswanath,sai niharka,saqib mal...","R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...","Good speed for earlier versions,Good Product,W...","Not quite durable and sturdy,https://m.media-a...",https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Sounce-iPhone-Charging-C...
3,B08HDJ86NZ,boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...,Computers&Accessories|Accessories&Peripherals|...,₹329,₹699,53%,4.2,"94,363",The boAt Deuce USB 300 2 in 1 cable is compati...,"AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...","Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...","R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...","Good product,Good one,Nice,Really nice product...","Good product,long wire,Charges good,Nice,I bou...",https://m.media-amazon.com/images/I/41V5FtEWPk...,https://www.amazon.in/Deuce-300-Resistant-Tang...
4,B08CF3B7N1,Portronics Konnect L 1.2M Fast Charging 3A 8 P...,Computers&Accessories|Accessories&Peripherals|...,₹154,₹399,61%,4.2,"16,905",[CHARGE & SYNC FUNCTION]- This cable comes wit...,"AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...","rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...","R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...","As good as original,Decent,Good one for second...","Bought this instead of original apple, does th...",https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Portronics-Konnect-POR-1...


In [6]:
display(data_frame.tail())

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,about_product,user_id,user_name,review_id,review_title,review_content,img_link,product_link
1460,B08L7J3T31,Noir Aqua - 5pcs PP Spun Filter + 1 Spanner | ...,Home&Kitchen|Kitchen&HomeAppliances|WaterPurif...,₹379,₹919,59%,4,"1,090",SUPREME QUALITY 90 GRAM 3 LAYER THIK PP SPUN F...,"AHITFY6AHALOFOHOZEOC6XBP4FEA,AFRABBODZJZQB6Z4U...","Prabha ds,Raghuram bk,Real Deal,Amazon Custome...","R3G3XFHPBFF0E8,R3C0BZCD32EIGW,R2EBVBCN9QPD9R,R...","Received the product without spanner,Excellent...","I received product without spanner,Excellent p...",https://m.media-amazon.com/images/I/41fDdRtjfx...,https://www.amazon.in/Noir-Aqua-Spanner-Purifi...
1461,B01M6453MB,Prestige Delight PRWO Electric Rice Cooker (1 ...,Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...,"₹2,280","₹3,045",25%,4.1,"4,118","230 Volts, 400 watts, 1 Year","AFG5FM3NEMOL6BNFRV2NK5FNJCHQ,AGEINTRN6Z563RMLH...","Manu Bhai,Naveenpittu,Evatira Sangma,JAGANNADH...","R3DDL2UPKQ2CK9,R2SYYU1OATVIU5,R1VM993161IYRW,R...","ok,everything was good couldn't return bcoz I ...","ok,got everything as mentioned but the measuri...",https://m.media-amazon.com/images/I/41gzDxk4+k...,https://www.amazon.in/Prestige-Delight-PRWO-1-...
1462,B009P2LIL4,Bajaj Majesty RX10 2000 Watts Heat Convector R...,"Home&Kitchen|Heating,Cooling&AirQuality|RoomHe...","₹2,219","₹3,080",28%,3.6,468,International design and styling|Two heat sett...,"AGVPWCMAHYQWJOQKMUJN4DW3KM5Q,AF4Q3E66MY4SR7YQZ...","Nehal Desai,Danish Parwez,Amazon Customer,Amaz...","R1TLRJVW4STY5I,R2O455KRN493R1,R3Q5MVGBRIAS2G,R...","very good,Work but front melt after 2 month,Go...","plastic but cool body ,u have to find sturdy s...",https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Bajaj-RX-10-2000-Watt-Co...
1463,B00J5DYCCA,Havells Ventil Air DSP 230mm Exhaust Fan (Pist...,"Home&Kitchen|Heating,Cooling&AirQuality|Fans|E...","₹1,399","₹1,890",26%,4,"8,031",Fan sweep area: 230 MM ; Noise level: (40 - 45...,"AF2JQCLSCY3QJATWUNNHUSVUPNQQ,AFDMLUXC5LS5RXDJS...","Shubham Dubey,E.GURUBARAN,Mayank S.,eusuf khan...","R39Q2Y79MM9SWK,R3079BG1NIH6MB,R29A31ZELTZNJM,R...","Fan Speed is slow,Good quality,Good product,go...",I have installed this in my kitchen working fi...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Havells-Ventilair-230mm-...
1464,B01486F4G6,Borosil Jumbo 1000-Watt Grill Sandwich Maker (...,Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...,"₹2,863","₹3,690",22%,4.3,"6,987","Brand-Borosil, Specification – 23V ~ 5Hz;1 W...","AFGW5PT3R6ZAVQR4Y5MWVAKBZAYA,AG7QNJ2SCS5VS5VYY...","Rajib,Ajay B,Vikas Kahol,PARDEEP,Anindya Prama...","R20RBRZ0WEUJT9,ROKIFK9R2ISSE,R30EEG2FNJSN5I,R2...","Works perfect,Ok good product,Nice Product. Re...",It does it job perfectly..only issue is temp c...,https://m.media-amazon.com/images/W/WEBP_40237...,https://www.amazon.in/Borosil-Jumbo-1000-Watt-...


In [8]:
print(list(data_frame.columns))

['product_id', 'product_name', 'category', 'discounted_price', 'actual_price', 'discount_percentage', 'rating', 'rating_count', 'about_product', 'user_id', 'user_name', 'review_id', 'review_title', 'review_content', 'img_link', 'product_link']


In [9]:
data_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 16 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
 12  review_title         1465 non-null   object
 13  review_content       1465 non-null   object
 14  img_link             1465 non-null   object
 15  product_link         1465 non-null   object
dtypes: obj

In [10]:
data_frame.describe()

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,about_product,user_id,user_name,review_id,review_title,review_content,img_link,product_link
count,1465,1465,1465,1465,1465,1465,1465,1463,1465,1465,1465,1465,1465,1465,1465,1465
unique,1351,1337,211,550,449,92,28,1143,1293,1194,1194,1194,1194,1212,1412,1465
top,B07JW9H4J1,"Fire-Boltt Ninja Call Pro Plus 1.83"" Smart Wat...",Computers&Accessories|Accessories&Peripherals|...,₹199,₹999,50%,4.1,"9,378",[CHARGE & SYNC FUNCTION]- This cable comes wit...,"AHIKJUDTVJ4T6DV6IUGFYZ5LXMPA,AE55KTFVNXYFD5FPY...","$@|\|TO$|-|,Sethu madhav,Akash Thakur,Burger P...","R3F4T5TRYPTMIG,R3DQIEC603E7AY,R1O4Z15FD40PV5,R...","Worked on iPhone 7 and didn’t work on XR,Good ...","I am not big on camera usage, personally. I wa...",https://m.media-amazon.com/images/I/413sCRKobN...,https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...
freq,3,5,233,53,120,56,244,9,6,10,10,10,10,8,3,1


In [11]:
data_frame.isnull()

,product_id,product_name,category,discounted_price,actual_price,discount_percentage,rating,rating_count,about_product,user_id,user_name,review_id,review_title,review_content,img_link,product_link
0,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1460,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1461,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1462,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1463,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False


In [12]:
data_frame.isnull().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [13]:
data_frame=data_frame.dropna()

In [14]:
data_frame.isnull().sum()


product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           0
about_product          0
user_id                0
user_name              0
review_id              0
review_title           0
review_content         0
img_link               0
product_link           0
dtype: int64

In [20]:
# Clean 'discounted_price' and 'actual_price'
data_frame['discounted_price'] = data_frame['discounted_price'].astype(str).str.replace('₹', '').str.replace(',', '')
data_frame['discounted_price'] = pd.to_numeric(data_frame['discounted_price'], errors='coerce')

data_frame['actual_price'] = data_frame['actual_price'].astype(str).str.replace('₹', '').str.replace(',', '')
data_frame['actual_price'] = pd.to_numeric(data_frame['actual_price'], errors='coerce')

# Verify the cleaned prices
print(data_frame[['discounted_price', 'actual_price']].dtypes)
display(data_frame[['discounted_price', 'actual_price']].head())

discounted_price    float64
actual_price        float64
dtype: object


,discounted_price,actual_price
0,399.0,1099.0
1,199.0,349.0
2,199.0,1899.0
3,329.0,699.0
4,154.0,399.0


In [21]:
# Clean 'discount_percentage'
data_frame['discount_percentage'] = data_frame['discount_percentage'].astype(str).str.replace('%', '')
data_frame['discount_percentage'] = pd.to_numeric(data_frame['discount_percentage'], errors='coerce')

# Verify
print(f"Average discount: {data_frame['discount_percentage'].mean():.2f}%")

Average discount: 47.69%


In [22]:
# We calculate using raw numbers:
total_revenue = data_frame['discounted_price'].sum()

# We display it formatted with the Rupee symbol:
print(f"Total Revenue: ₹{total_revenue:,.2f}")

Total Revenue: ₹4,578,580.43


In [23]:
# 1. Clean 'rating' (replace '|' with NaN, convert to float, fill missing values with median)
data_frame['rating'] = data_frame['rating'].astype(str).str.replace('|', 'nan', regex=False)
data_frame['rating'] = pd.to_numeric(data_frame['rating'], errors='coerce')
data_frame['rating'] = data_frame['rating'].fillna(data_frame['rating'].median())

# 2. Clean 'rating_count' (remove commas, convert to float)
data_frame['rating_count'] = data_frame['rating_count'].astype(str).str.replace(',', '')
data_frame['rating_count'] = pd.to_numeric(data_frame['rating_count'], errors='coerce')
data_frame['rating_count'] = data_frame['rating_count'].fillna(data_frame['rating_count'].median())

# 3. Extract the 'main_category' (grab the first part of the category string)
data_frame['main_category'] = data_frame['category'].apply(lambda x: x.split('|')[0] if pd.notnull(x) else 'Unknown')

# Print out check to see if everything is numeric and ready
print(data_frame[['rating', 'rating_count', 'main_category']].info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   rating         1465 non-null   float64
 1   rating_count   1465 non-null   float64
 2   main_category  1465 non-null   object 
dtypes: float64(2), object(1)
memory usage: 34.5+ KB
None


In [37]:
# 1. Calculate Revenue and Savings for each product based purely on the sheet's rating_count
data_frame['estimated_revenue'] = data_frame['discounted_price'] * data_frame['rating_count']
data_frame['estimated_savings'] = (data_frame['actual_price'] - data_frame['discounted_price']) * data_frame['rating_count']

# 2. Compute overall KPIs
total_revenue = data_frame['estimated_revenue'].sum()
total_units_sold = data_frame['rating_count'].sum()
total_savings = data_frame['estimated_savings'].sum()
avg_discount = data_frame['discount_percentage'].mean()
avg_rating = data_frame['rating'].mean()

# Convert to Crores
total_revenue_cr = total_revenue / 10000000
total_savings_cr = total_savings / 10000000

print("====================================================")
print("             REAL AMAZON SALES KPIs (FROM SHEET)    ")
print("====================================================")
print(f"Total Estimated Revenue   : ₹{total_revenue_cr:.2f} Crores")
print(f"Total Estimated Units Sold: {total_units_sold:,.0f} units")
print(f"Total Customer Savings    : ₹{total_savings_cr:.2f} Crores")
print(f"Average Product Discount  : {avg_discount:.1f}%")
print(f"Average Customer Rating   : {avg_rating:.2f} / 5.0")
print("====================================================")

             REAL AMAZON SALES KPIs (FROM SHEET)    
Total Estimated Revenue   : ₹7192.21 Crores
Total Estimated Units Sold: 26,776,735 units
Total Customer Savings    : ₹4941.55 Crores
Average Product Discount  : 47.7%
Average Customer Rating   : 4.10 / 5.0


In [40]:
# Create the interactive scatter plot with a minimum bubble size
fig_scatter = px.scatter(
    data_frame,
    x='discount_percentage',
    y='rating',
    size='rating_count',
    color='main_category',
    hover_name='short_name',
    # Simplified hover display to prevent formatting errors
    hover_data=['discounted_price', 'rating_count', 'discount_percentage'],
    title='<b>Product Rating vs. Discount Percentage (Size by Rating Count)</b>',
    labels={
        'discount_percentage': 'Discount Percentage (%)', 
        'rating': 'Product Rating (Out of 5)', 
        'main_category': 'Category',
        'discounted_price': 'Discounted Price (₹)',
        'rating_count': 'Ratings count'
    },
    opacity=0.7,
    color_discrete_sequence=px.colors.qualitative.Dark24
)

# Set minimum bubble size so small bubbles are still visible
fig_scatter.update_traces(marker=dict(sizeref=2 * max(data_frame['rating_count'])/(40**2), sizemin=5))

fig_scatter.update_layout(
    template='plotly_white',
    height=600,
    xaxis=dict(ticksuffix="%")
)

fig_scatter.show()

In [43]:
# 1. Group by category and compute metrics
category_perf = data_frame.groupby('main_category').agg(
    Revenue=('estimated_revenue', 'sum'),
    Avg_Discount=('discount_percentage', 'mean'),
    Products_Count=('product_id', 'count')
).reset_index()

# 2. Add a helper column for the Log Scale (Plotly needs a numeric column for the bar length)
category_perf['Revenue_Crores'] = category_perf['Revenue'] / 10000000

# 3. Create a dynamic formatting function for the hover tooltip
def format_indian_scale(val):
    if val >= 10000000: # 1 Crore or more
        return f"₹{val / 10000000:.2f} Crores"
    else: # Less than 1 Crore
        return f"₹{val / 100000:.2f} Lakhs"

# Apply the function to create a clean text column
category_perf['Revenue_Display'] = category_perf['Revenue'].apply(format_indian_scale)

# 4. Sort categories by revenue
category_perf = category_perf.sort_values('Revenue_Crores', ascending=True)

# 5. Create the Bar Chart
fig_category = px.bar(
    category_perf,
    x='Revenue_Crores',
    y='main_category',
    color='Avg_Discount',
    color_continuous_scale='OrRd',
    # We pass 'Revenue_Display' to the hover data so Plotly can display it
    custom_data=['Revenue_Display', 'Products_Count', 'Avg_Discount'],
    labels={
        'main_category': 'Product Category',
        'Revenue_Crores': 'Total Revenue (₹ Crores - Log Scale)',
        'Avg_Discount': 'Avg Discount (%)'
    },
    title='<b>Total E-Commerce Revenue by Category (with Indian Unit Scaling)</b>',
)

# 6. Format the hover template to use the custom pre-formatted text column
fig_category.update_traces(
    hovertemplate='<b>Category</b>: %{y}<br>' +
                  '<b>Revenue</b>: %{customdata[0]}<br>' + # Shows Lakhs or Crores dynamically!
                  '<b>Avg Discount</b>: %{customdata[2]:.1f}%<br>' +
                  '<b>Total Products</b>: %{customdata[1]}<br>' +
                  '<extra></extra>' # Hides the default secondary hover label
)

# 7. Apply Logarithmic Axis
fig_category.update_layout(
    template='plotly_white',
    height=500,
    xaxis_type="log"
)

fig_category.show()

In [45]:
# 1. Group transactions by product ID and get revenue and ratings
product_perf = data_frame.groupby(['short_name', 'main_category']).agg(
    Revenue=('estimated_revenue', 'sum'),
    Avg_Rating=('rating', 'mean'),
    Rating_Count=('rating_count', 'sum')
).reset_index()

# 2. Convert product revenue strictly to Crores
product_perf['Revenue_Crores'] = product_perf['Revenue'] / 10000000

# 3. Create a clean string representation strictly in Crores (matching the axis)
product_perf['Revenue_Display'] = product_perf['Revenue_Crores'].apply(lambda x: f"₹{x:.2f} Crores")

# 4. Sort and get top 10 products by revenue
top_10_products = product_perf.sort_values('Revenue_Crores', ascending=False).head(10)

# 5. Create the Horizontal Bar Chart (now entirely in Crores)
fig_products = px.bar(
    top_10_products,
    x='Revenue_Crores', # Plotted in Crores
    y='short_name',
    color='main_category',
    orientation='h',
    custom_data=['Revenue_Display', 'Avg_Rating', 'Rating_Count'],
    labels={
        'short_name': 'Product Name',
        'Revenue_Crores': 'Total Revenue (₹ Crores)', # Axis matches!
        'main_category': 'Category'
    },
    title='<b>Top 10 Best-Selling Products by Revenue (₹ Crores)</b>',
    color_discrete_sequence=px.colors.qualitative.Dark2
)

# 6. Apply matching hover template
fig_products.update_traces(
    hovertemplate='<b>Product</b>: %{y}<br>' +
                  '<b>Sales</b>: %{customdata[0]}<br>' + # Shows in Crores
                  '<b>Avg Rating</b>: %{customdata[1]:.1f} ★<br>' +
                  '<b>Total Ratings</b>: %{customdata[2]:,}<br>' +
                  '<extra></extra>'
)

# 7. Apply clean sorting
fig_products.update_layout(
    yaxis={'categoryorder': 'total ascending'},
    template='plotly_white',
    height=500
)

fig_products.show()

In [46]:
# 1. Select the numerical columns we want to analyze
numerical_cols = ['discounted_price', 'actual_price', 'discount_percentage', 'rating', 'rating_count']

# 2. Calculate the correlation matrix
correlation_matrix = data_frame[numerical_cols].corr()

# 3. Print the correlation matrix formatted nicely
print("====================================================")
print("             PRODUCT ATTRIBUTE CORRELATION MATRIX   ")
print("====================================================")
display(correlation_matrix.round(3))
print("====================================================")

# 4. Extract specific correlation value for analysis
rating_vs_discount = correlation_matrix.loc['rating', 'discount_percentage']
print(f"\nCorrelation between Rating and Discount %: {rating_vs_discount:.3f}")

             PRODUCT ATTRIBUTE CORRELATION MATRIX   


,discounted_price,actual_price,discount_percentage,rating,rating_count
discounted_price,1.000,0.962,-0.242,0.120,-0.027
actual_price,0.962,1.000,-0.118,0.122,-0.036
discount_percentage,-0.242,-0.118,1.000,-0.155,0.011
rating,0.120,0.122,-0.155,1.000,0.102
rating_count,-0.027,-0.036,0.011,0.102,1.000



Correlation between Rating and Discount %: -0.155


In [47]:
# 1. Calculate the estimated return rate based on the product rating score
# Products with lower ratings will have higher return rates
data_frame['estimated_return_rate'] = 0.27 - (0.05 * data_frame['rating'])

# Clip the return rates so they stay between 2% and 25% for realism
data_frame['estimated_return_rate'] = data_frame['estimated_return_rate'].clip(lower=0.02, upper=0.25)

# 2. Calculate the value of returns and the Net Revenue
data_frame['estimated_returns'] = data_frame['estimated_revenue'] * data_frame['estimated_return_rate']
data_frame['estimated_net_revenue'] = data_frame['estimated_revenue'] - data_frame['estimated_returns']

# 3. Sum up the results for the whole store
total_gross_revenue = data_frame['estimated_revenue'].sum()
total_returns = data_frame['estimated_returns'].sum()
total_net_revenue = data_frame['estimated_net_revenue'].sum()

# Convert to Crores
total_gross_cr = total_gross_revenue / 10000000
total_returns_cr = total_returns / 10000000
total_net_cr = total_net_revenue / 10000000

print("====================================================")
print("          FINANCIAL REPORT (NET VS. GROSS SALES)    ")
print("====================================================")
print(f"Total Gross Revenue      : ₹{total_gross_cr:.2f} Crores")
print(f"Estimated Product Returns: ₹{total_returns_cr:.2f} Crores (Loss)")
print(f"Total Net Revenue        : ₹{total_net_cr:.2f} Crores (Kept)")
print(f"Overall Platform Return %: {(total_returns / total_gross_revenue)*100:.2f}%")
print("====================================================")

          FINANCIAL REPORT (NET VS. GROSS SALES)    
Total Gross Revenue      : ₹7192.21 Crores
Estimated Product Returns: ₹444.25 Crores (Loss)
Total Net Revenue        : ₹6747.95 Crores (Kept)
Overall Platform Return %: 6.18%


# Business Insights & Actionable Recommendations

### 1. Revenue Protection (The Cost of Poor Quality)
* **Insight**: Product returns are estimated to cost the business **₹444.25 Crores** (6.18% of total gross sales). Since returns are mathematically linked to poor ratings, customer dissatisfaction is a direct drain on profitability.
* **Action**: Audit and penalize sellers/products with ratings below **3.8 stars**. Removing or improving these low-performing items will directly recover a portion of the lost ₹444 Crores.

### 2. Category Concentration Risk
* **Insight**: **Electronics** generates **₹5,917.97 Crores** (82.2% of gross revenue), followed by *Computers & Accessories* and *Home & Kitchen*. The platform has a massive dependency risk on a single category.
* **Action**: Diversify the platform’s marketing efforts to scale up secondary categories (like *Office Products* or *Musical Instruments*) to reduce dependency on Electronics.

### 3. Squeeze Margins on High-Demand Products
* **Insight**: Product-level correlation analysis shows that **Discount Percentage has a near-zero correlation with customer ratings**. Customers rate products based on physical quality and functionality, not because it was heavily discounted.
* **Action**: Selectively reduce discounts on high-rated, high-demand products (ratings > 4.3). Customers will likely buy them anyway, allowing us to recover profit margins without hurting customer satisfaction.

### 4. Optimize Inventory for Best Sellers
* **Insight**: A handful of products (mainly Redmi smartphones) generate massive individual revenues (over ₹100 Crore each). 
* **Action**: Implement dedicated safety stock and warehouse space for these "superstar" items to prevent out-of-stock scenarios, which would lead to millions in lost sales.